### DISCUSSION TITLE
[Service Bus] Retries and Dead Letter Queue

### DISCUSSION THREAD

**Discussion by schause** [Microsoft/spring-cloud-azure#1055] [Q&A] [answered]
Hi all,
im trying something (presumably) very simple - on specific exceptions/errors that occur when handling a servicebus message, i want either to have exponential fallback (with scheduled delivery) or to move it directly into the DLQ when i know that the message will never be resolved by retries.
However i did quite some time of research and trial/error and im not able to make it work.

Current Setup is this: I have a `Consumer<Message<MyMessage>>`  and i extract a `com.azure.spring.messaging.checkpoint.Checkpointer` from the `MessageHeaders`. This Checkpointer has only two methods, success and failure. It seems that i can neither reschedule the message with a fallback delay nor that i can say: please dont re-send this, move it directly to DLQ.
As i couldn find a way to make it work, i also have tried to do a hacky approach and succeed the message and send a new copy of the message to DLQ wiith the underlying `com.azure.messaging.servicebus.ServiceBusSenderAsyncClient`. however, it also offers me no possibility to send to dead letter queue.
The docs of .net client show that there is a queue client with deadLetterAsync method, thats seems also not to be available in Java: https://learn.microsoft.com/en-us/dotnet/api/microsoft.servicebus.messaging.queueclient.deadletterasync?view=azure-dotnet#microsoft-servicebus-messaging-queueclient-deadletterasync(system-guid-system-string-system-string)

Can you give me any hints on how to get this working? This seems to be a not so exotic problem and im curious if im missing something thats super obvious.

Please find my dependency tree below if thats needed:

\- com.azure.spring:spring-cloud-azure-stream-binder-servicebus:jar:4.7.0:compile
[INFO] |     +- com.azure.spring:spring-cloud-azure-stream-binder-servicebus-core:jar:4.7.0:compile
[INFO] |     |  +- org.springframework.cloud:spring-cloud-stream:jar:3.2.5:compile
[INFO] |     |  +- com.azure.spring:spring-integration-azure-servicebus:jar:4.7.0:compile
[INFO] |     |  |  +- com.azure.spring:spring-integration-azure-core:jar:4.7.0:compile
[INFO] |     |  |  |  \- com.azure.spring:spring-messaging-azure:jar:4.7.0:compile
[INFO] |     |  |  \- com.azure.spring:spring-messaging-azure-servicebus:jar:4.7.0:compile
[INFO] |     |  \- com.azure:azure-messaging-servicebus:jar:7.13.3:compile

**Comment by saragluna** [comment, marked as answer]
Hi @schause, please check this https://github.com/Azure/azure-sdk-for-java/blob/5bc550c9a5de4f8ee93a5b3141500ee34be3850d/sdk/spring/spring-cloud-azure-stream-binder-servicebus/src/main/java/com/azure/spring/cloud/stream/binder/servicebus/ServiceBusMessageChannelBinder.java#L170-L193 and https://github.com/Azure/azure-sdk-for-java/pull/14861 to see if it can work?
